src/bindings/python/src/pyopenvino/graph/discrete_type_info.hpp

```diff
@@ -6,6 +6,26 @@
 
 #include <pybind11/pybind11.h>
 
+#include "openvino/core/type.hpp"
+#include <string>
 namespace py = pybind11;
 
+
+// DiscreteTypeInfo doesn't own provided memory. Wrapper allows to avoid leaks.
+class DiscreteTypeInfoWrapper : public ov::DiscreteTypeInfo {
+private:
+    const std::string name_str;
+    const std::string version_id_str;
+
+public:
+    DiscreteTypeInfoWrapper(std::string _name_str, std::string _version_id_str)
+        : DiscreteTypeInfo(nullptr, nullptr, nullptr),
```

Comment:
The implementation of the constructor can also stay in [discrete_type_info.cpp](https://github.com/openvinotoolkit/openvino/pull/27664/files#diff-b45a18057e412cc666e99a78a7ac081c53859a5b8a5e527ba2fa294f7fb2930d)

Reply:
In this particular case it doesn't make a lot of sense -- `DiscreteTypeInfoWrapper` class contains only one ctor here.

Reply:
I was thinking about sth like this: [header](https://github.com/openvinotoolkit/openvino/blob/313e39e50b98a057ffebf2f5c553636305923661/src/bindings/js/node/include/core_wrap.hpp#L18), [source file](https://github.com/openvinotoolkit/openvino/blob/313e39e50b98a057ffebf2f5c553636305923661/src/bindings/js/node/src/core_wrap.cpp#L44)